services/src/test-library/src/main/java/org/openkilda/testing/service/lockkeeper/LockKeeperVirtualImpl.java

```diff
@@ -61,46 +69,56 @@ public LockKeeperVirtualImpl() throws DockerCertificateException, DockerExceptio
+        restTemplate.exchange("/flows", HttpMethod.POST,
+                new HttpEntity<>(flows, buildJsonHeaders()), String.class);
+        LOGGER.debug("Added flows: {}", flows.stream()
                 .map(flow -> String.format("%s->%s", flow.getInPort(), flow.getOutPort()))
                 .collect(Collectors.toList()));
     }
 
     @Override
     public void removeFlows(List<ASwitchFlow> flows) {
-        flows.forEach(flow -> mininet.removeFlow(ASWITCH_NAME, flow.getInPort()));
-        log.debug("Removed flows: {}", flows.stream()
+        restTemplate.exchange("/flows", HttpMethod.DELETE,
+                new HttpEntity<>(flows, buildJsonHeaders()), String.class);
+        LOGGER.debug("Removed flows: {}", flows.stream()
                 .map(flow -> String.format("%s->%s", flow.getInPort(), flow.getOutPort()))
                 .collect(Collectors.toList()));
     }
 
     @Override
     public List<ASwitchFlow> getAllFlows() {
-        throw new UnsupportedOperationException("getAllFlows operation for a-switch is not available on virtual env");
+        ASwitchFlow[] flows = restTemplate.exchange("/flows", HttpMethod.GET,
+                new HttpEntity(buildJsonHeaders()), ASwitchFlow[].class).getBody();
+        return Arrays.asList(flows);
     }
 
     @Override
     public void portsUp(List<Integer> ports) {
-        ports.forEach(port -> mininet.portUp(ASWITCH_NAME, port));
+        restTemplate.exchange("/ports", HttpMethod.POST,
+                new HttpEntity<>(ports, buildJsonHeaders()), String.class);
+        LOGGER.debug("Brought up ports: {}", ports);
     }
 
     @Override
     public void portsDown(List<Integer> ports) {
-        ports.forEach(port -> mininet.portDown(ASWITCH_NAME, port));
+        restTemplate.exchange("/ports", HttpMethod.DELETE,
+                new HttpEntity<>(ports, buildJsonHeaders()), String.class);
+        LOGGER.debug("Brought down ports: {}", ports);
     }
 
     @Override
-    public void knockoutSwitch(SwitchId switchId) {
-        mininet.knockoutSwitch(topology.getSwitches().stream()
-                .filter(sw -> sw.getDpId().equals(switchId)).findFirst().get().getName());
+    public void knockoutSwitch(Switch switchDef) {
+        restTemplate.exchange("/knockoutswitch", HttpMethod.POST,
+                new HttpEntity<>(new SwitchModify(switchDef.getName(), null), buildJsonHeaders()), String.class);
+        LOGGER.debug("Knocking out switch: {}", switchDef.getName());
     }
 
     @Override
-    public void reviveSwitch(SwitchId switchId) {
-        String switchName = topology.getSwitches().stream()
-                .filter(sw -> sw.getDpId().equals(switchId)).findFirst().get().getName();
-        mininet.revive(switchName, VIRTUAL_CONTROLLER_ADDRESS);
+    public void reviveSwitch(Switch switchDef) {
+        restTemplate.exchange("/reviveswitch", HttpMethod.POST,
+                new HttpEntity<>(new SwitchModify(switchDef.getName(), VIRTUAL_CONTROLLER_ADDRESS),
```

Comment:
since lab-service will run on a remote server I suppose it won't be able to resolve the "tcp:kilda:6653" host. So we should instead pass executor's IP address (which will have a Floodlight and the whole Kilda running locally) from tests, which can then be used at the lab-service's side